mempool/src/verify.rs

```diff
@@ -123,12 +138,12 @@ pub(crate) async fn verify_tx<'a>(
 
     if !transaction.is_valid_at(block_height) {
         log::debug!("Transaction invalid at block {}", block_height);
-        return Err(VerifyErr::Invalid);
+        return Err(VerifyErr::InvalidBlockHeight);
     }
 
     if blockchain.contains_tx_in_validity_window(&transaction.hash(), None) {
         log::debug!("Transaction has already been mined");
-        return Err(VerifyErr::Invalid);
+        return Err(VerifyErr::InvalidTxWindow);
```

Comment:
When I first saw the `InvalidTxWindow` error, I thought it would be the error that the transaction is invalid because of block height (the window already passed, or hasn't started). I see now that that would be the `InvalidBlockHeight` error.

The naming might be confusing, because an invalid block height is an invalid tx window at the same time.

How about renaming `VerifyErr::InvalidTxWindow` => `VerifyErr::AlreadyMined`, or `AlreadyIncluded`?